<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <title>Access precomputed correlations &middot; Sina Rüeger</title>
        <meta name="description" content="LD Part 1">
        <meta name="HandheldFriendly" content="True">
        <meta name="MobileOptimized" content="320">
        <meta name="generator" content="Hugo 0.38.2" />
        <meta name="robots" content="index,follow">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <link rel="stylesheet" href="/dist/styles.css">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Anaheim|Assistant|Open+Sans:300italic,400italic,600italic,700italic,400,600,700,300&subset=latin,cyrillic-ext,latin-ext,cyrillic">
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous">
        
        
    </head>
    <body>
        

        <div id="wrapper">
            <header class="site-header">
                <div class="container">
                    <div class="site-title-wrapper">
                        
                            <h1 class="site-title">
                                <a title="Analysing genetic data" href="/">Analysing genetic data</a>
                            </h1>
                        
                        <a class="button-square" href="/index.xml"><i class="fa fa-rss"></i></a>
                        
                            <a class="button-square button-social hint--top" data-hint="Twitter" title="Twitter" href="https://twitter.com/sinarueeger">
                                <i class="fa fa-twitter"></i>
                            </a>
                        
                        
                        
                            <a class="button-square button-social hint--top" data-hint="Github" title="Github" href="https://github.com/sinarueeger">
                                <i class="fa fa-github"></i>
                            </a>
                        
                        
                        
                            <a class="button-square button-social hint--top" data-hint="LinkedIn" title="LinkedIn" href="https://www.linkedin.com/in/sinarueeger/">
                                <i class="fa fa-linkedin"></i>
                            </a>
                        
                        
                        
                            <a class="button-square button-social hint--top" data-hint="Google Scholar" title="Google Scholar" href="https://scholar.google.com/citations?view_op=list_works&amp;hl=de&amp;authuser=1&amp;user=OD9gCh4AAAAJ">
                                <i class="fa fa-book"></i>
                            </a>
                        
                        
                            <a class="button-square button-social hint--top" data-hint="Email" title="Email" href="mailto:sina.rueeger@gmail.com">
                                <i class="fa fa-envelope"></i>
                            </a>
                        
                    </div>

                    <ul class="site-nav">
                        
    <li class="site-nav-item">
        <a title="Blog" href="/post/">Blog</a>
    </li>

    <li class="site-nav-item">
        <a title="Resources" href="/page/resources/">Resources</a>
    </li>

    <li class="site-nav-item">
        <a title="About" href="/page/about/">About</a>
    </li>

    <li class="site-nav-item">
        <a title="Contact" href="/page/contact/">Contact</a>
    </li>

                    </ul>
                </div>
            </header>

            <div id="container">


<div class="container">
    <article class="post-container" itemscope="" itemtype="http://schema.org/BlogPosting">
        <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Access precomputed correlations</h1>
    
        <p class="post-description" itemprop="description">LD Part 1</p>
    
    <p class="post-date">
        <span>Published <time datetime="2018-08-09" itemprop="datePublished">Thu, Aug 9, 2018</time></span>
        <span>by</span>
        <span itemscope="" itemprop="author" itemtype="https://schema.org/Person">
            <span itemprop="name">
                <a href="https://twitter.com/sinarueeger" itemprop="url" rel="author">Sina Rüeger</a>
            </span>
        </span>
    </p>
</header>

        <div class="post-content clearfix" itemprop="articleBody">
    

    <link href="/rmarkdown-libs/font-awesome/css/fontawesome-all.min.css" rel="stylesheet" />


<p>A weirdly difficult thing to accomplish, is, to calculate the correlations between genetic markers and store these in a matrix. Thing is - correlations have to be computed <em>ALL THE TIME</em> for many statistical methods in genetics. And in this blogpost I am exploring how to do this efficiently.</p>
<!---------- ------------------------------->
<div id="goal---what-do-we-want" class="section level2">
<h2>Goal - what do we want?</h2>
<!---------- ------------------------------->
<p>At the end, we want to know the genetic correlation between two or more markers in a specific human population.</p>
<!---------- ------------------------------->
</div>
<div id="two-approaches" class="section level2">
<h2>Two approaches</h2>
<!---------- ------------------------------->
<p>There are two ways of doing this:</p>
<ol style="list-style-type: decimal">
<li>Download the genetic data from which you want to estimate your correlations + calculate the correlations using some efficient approach.</li>
<li>Access precomputed correlations.</li>
</ol>
<p>Approach 1. is flexible and can be quickly updated to new reference panels. Downside: requires large computation and storage space (e.g. 1KG is xx MB). Recomputing might not be always necessary. Approach 2. does not require large computation and storage space. Downside: limited to certain ranges of data. Limited to older reference panels.</p>
<p>Approach 2 is suitable if small cor matrix is needed (limited range). Approach 1 is suitable, if large matrix is needed.</p>
<table style="width:100%;">
<colgroup>
<col width="9%" />
<col width="22%" />
<col width="39%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th>Approach <i class="fas  fa-wrench "></i></th>
<th>Advantages <i class="fas  fa-smile "></i></th>
<th>Downsides <i class="fas  fa-frown "></i></th>
<th>Useful when…</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>(1) Local computation of LD</td>
<td>flexible and can be quickly updated to new reference panels</td>
<td>Requires large computation and storage space (e.g. 1KG is xx MB). Recomputing might not be always necessary.</td>
<td>- large correlation matrix is needed - when special reference panel is needed</td>
</tr>
<tr class="even">
<td>(2) Access precomputed LD</td>
<td>does not require large computation and storage space.</td>
<td>limited to certain ranges of data. Limited to older reference panels.</td>
<td>small cor matrix is needed (limited range)</td>
</tr>
</tbody>
</table>
<p>I will be focussing for now on approach 2, and then explore approach 1 in a next blogpost.</p>
<p>Spoileralert: Using approach 2 does not get you far. not easy. only 1 snp vs many. Limited to 500 KB. Maybe a hack? (locuszoomplot)</p>
<!---------- ------------------------------->
</div>
<div id="our-toy-data" class="section level2">
<h2>Our toy data</h2>
<!---------- ------------------------------->
<p>We will use data from the <a href="https://sinarueeger.github.io/2018/07/30/locuszoomplot/">previous blogpost</a>, where the focus was on extracting annotation using the package <code>biomaRt</code>. In this blogpost, we will complete that locuszoomplot by adding the LD information.</p>
<pre class="r"><code>## Data Source URL
url &lt;- &quot;https://portals.broadinstitute.org/collaboration/giant/images/2/21/BMI_All_ancestry.fmt.gzip&quot;

#url &lt;- &quot;jenger.riken.jp/1analysisresult_qtl_download/All_2017_BMI_BBJ_autosome.txt.gz&quot;

## Import BMI summary statistics dat.bmi &lt;- read_tsv(file = url) ##
## taking too long, let&#39;s use fread instead.

dat.bmi &lt;- data.table::fread(url, verbose = FALSE)

## Rename some columns
dat.bmi &lt;- dat.bmi %&gt;% rename(SNP = SNPNAME, P = Pvalue)

## Extract region
dat.bmi.sel &lt;- dat.bmi %&gt;% slice(which.min(P))
dat.bmi.sel

## range region
range &lt;- 5e+05
sel.chr &lt;- dat.bmi.sel$CHR
sel.pos &lt;- dat.bmi.sel$POS

dat.bmi.sel.region &lt;- dat.bmi %&gt;% filter(CHR == sel.chr, between(POS, sel.pos - 
      range, sel.pos + range))</code></pre>
<p>So what we are interested in, is the LD between our top SNP rs1421085 and all remaining SNPs. LD is typically measured either as the squred correlation (R2) or dprime (D’).</p>
<p>Remember that this dataset has positions on build GRCh37, while most databases are on build GRCh38 by now.</p>
<pre class="r"><code>library(rsnps)
snp &lt;- dat.bmi.sel$SNP
sm &lt;- ncbi_snp_summary(snp) %&gt;% separate(chrpos, c(&quot;chr&quot;, &quot;pos&quot;))
sel.pos == as.numeric(sm$pos)</code></pre>
<pre><code>## [1] FALSE</code></pre>
<!---------- ------------------------------->
</div>
<div id="kinda-works-httpsrest.ensembl.org" class="section level2">
<h2>Kinda works: <code>https://rest.ensembl.org</code></h2>
<!---------- ------------------------------->
<p>Let’s quickly repeat what our goal is:</p>
<p>Extract the correlation between SNPs of 1000 Genomes reference panel - without downloading any data, - fairly quick and - in R?</p>
<pre class="r"><code>snp &lt;- dat.bmi.sel$SNP
range.kb &lt;- range/1e3 ## in k bp</code></pre>
<p>To access the api at ensembl, we need the following three packages loaded.</p>
<pre class="r"><code>library(httr)
library(jsonlite)</code></pre>
<pre><code>## 
## Attaching package: &#39;jsonlite&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     flatten</code></pre>
<pre class="r"><code>library(xml2)</code></pre>
<div id="what-reference-panels-population-can-we-choose-from" class="section level3">
<h3>What reference panels / population can we choose from?</h3>
<!------------------------------>
<pre class="r"><code>server &lt;- &quot;https://rest.ensembl.org&quot;
ext &lt;- &quot;/info/variation/populations/homo_sapiens?filter=LD&quot;
 
r &lt;- GET(paste(server, ext, sep = &quot;&quot;), content_type(&quot;application/json&quot;))
 
stop_for_status(r)
 
# use this if you get a simple nested list back, otherwise inspect its structure
# head(data.frame(t(sapply(content(r),c))))
head(fromJSON(toJSON(content(r))))</code></pre>
<pre><code>##                      name
## 1 1000GENOMES:phase_3:ACB
## 2 1000GENOMES:phase_3:ASW
## 3 1000GENOMES:phase_3:BEB
## 4 1000GENOMES:phase_3:CDX
## 5 1000GENOMES:phase_3:CEU
## 6 1000GENOMES:phase_3:CHB
##                                                  description size
## 1                              African Caribbean in Barbados   96
## 2                           African Ancestry in Southwest US   61
## 3                                      Bengali in Bangladesh   86
## 4                        Chinese Dai in Xishuangbanna, China   93
## 5 Utah residents with Northern and Western European ancestry   99
## 6                               Han Chinese in Bejing, China  103</code></pre>
<p><code>name</code> is the key for the population. <code>size</code> refers to the number of individuals in the reference panel. Note that these are all populations with around 100 individuals (meaning, we are accessing correlation estimations with a large error margin). There are also the super population available, simply replace the last three characters by <code>EUR</code>, <code>AFR</code>, <code>AMR</code>, <code>EAS</code>, <code>SAS</code>.</p>
<p>And anyway - how do we know which population to pick? One way is, to read up what kind of individuals were present. In our case - mostly europeans.</p>
<p>We could also</p>
<p>Here is a better overview of the all populations, source code <a href="https://github.com/sinarueeger/map-1000genomes">here</a>.</p>
<iframe seamless src="/misc/map-1000genomes-populations.html" width="100%" height="500">
</iframe>
</div>
<div id="ld-between-snp1-and-region" class="section level3">
<h3>LD between SNP1 and region</h3>
<!------------------------------>
<p>This is from <a href="https://rest.ensembl.org/documentation/info/ld_id_get">here</a>.</p>
<p>The default window size is 500 kb. There are also thresholds for r2 (e.g. if you want to filter all SNPs with an r2 &gt; 0.8).</p>
<p>What is required is the SNP, the name string, here .</p>
<pre class="r"><code>server &lt;- &quot;https://rest.ensembl.org&quot;
ext &lt;- glue::glue(&quot;/ld/human/{snp}/1000GENOMES:phase_3:EUR?&quot;)
## Window size in kb. The maximum allowed value for the window size is 500 kb. LD is computed for the given variant and all variants that are located within the specified window.

r &lt;- GET(paste(server, ext, sep = &quot;&quot;), content_type(&quot;application/json&quot;))
stop_for_status(r)
LD.SNP.region &lt;- as_tibble(fromJSON(toJSON(content(r)))) %&gt;% unnest() %&gt;% mutate(r2 = as.numeric(r2))</code></pre>
<p>So <code>LD.snp.region</code> contains all SNPs in 1KG that were +/- 500 kb away from our top SNP. So thats, LD of one SNP with many other SNPs. But what if we want the correlation between all SNPs?</p>
</div>
<div id="calculate-ld-matrix" class="section level3">
<h3>Calculate LD matrix</h3>
<!------------------------------>
<p>This is what the following does. <a href="https://rest.ensembl.org/documentation/info/ld_region_get">here</a>.</p>
<p>We can calculate the LD matrix of a full region, max 1 Mb wide. Because of computation, we limit it to +/- 100 kb.</p>
<pre class="r"><code>## 
## example: ext &lt;- &quot;/ld/human/region/6:25837556..25843455/1000GENOMES:phase_3:KHV?&quot;
## Query region. A maximum of 1Mb is allowed.

ext &lt;- glue::glue(&quot;/ld/human/region/{sel.chr}:{sel.pos - range/10}..{sel.pos + range/10}/1000GENOMES:phase_3:EUR?&quot;)

r &lt;- GET(paste(server, ext, sep = &quot;&quot;), content_type(&quot;application/json&quot;))
stop_for_status(r)
LD.matrix &lt;- as_tibble(fromJSON(toJSON(content(r)))) %&gt;% unnest()
LD.matrix</code></pre>
<pre><code>## # A tibble: 31,874 x 5
##    variation1 population_name         variation2 r2       d_prime 
##    &lt;chr&gt;      &lt;chr&gt;                   &lt;chr&gt;      &lt;chr&gt;    &lt;chr&gt;   
##  1 rs8058930  1000GENOMES:phase_3:EUR rs62033437 0.078515 0.959980
##  2 rs9931164  1000GENOMES:phase_3:EUR rs74018596 0.606811 1.000000
##  3 rs10852521 1000GENOMES:phase_3:EUR rs56094641 0.650360 0.994565
##  4 rs9937429  1000GENOMES:phase_3:EUR rs11075995 0.994171 1.000000
##  5 rs17817288 1000GENOMES:phase_3:EUR rs9926289  0.630724 0.999999
##  6 rs62048375 1000GENOMES:phase_3:EUR rs73609995 0.960282 1.000000
##  7 rs12596393 1000GENOMES:phase_3:EUR rs17818902 0.365967 0.999975
##  8 rs73607075 1000GENOMES:phase_3:EUR rs7188250  0.050316 0.999928
##  9 rs28562191 1000GENOMES:phase_3:EUR rs4783819  0.428655 0.999996
## 10 rs72805632 1000GENOMES:phase_3:EUR rs16952581 0.143193 0.999972
## # ... with 31,864 more rows</code></pre>
<p>Make a traditional LD matrix plot (scale white to red).</p>
<pre class="r"><code>#ggplot(data = LD.matrix) + geom_tile(aes(x = variation1, y = variation2, fill = r2)) + scale_fill_gradient(low = &#39;white&#39;, high = &quot;red&quot;)</code></pre>
</div>
<div id="ld-between-snp1-and-other-snps" class="section level3">
<h3>LD between SNP1 and other SNPs</h3>
<!------------------------------>
<p><a href="https://rest.ensembl.org/documentation/info/ld_pairwise_get">Here</a> The third option is to give a set of SNP ids, and calculate the correlations among these. Implemented in ensembl is only the correlatino between two SNPs, so we will have to extend this to many SNPs.</p>
<pre class="r"><code>f.extract.ld &lt;- function(SNP.id2 = NULL, SNP.id1 = NULL, POP = NULL)
{
  ext &lt;- glue::glue(&quot;/ld/human/pairwise/{SNP.id1}/{SNP.id2}?population_name=1000GENOMES:phase_3:{POP}&quot;) 
  
  server &lt;- &quot;https://rest.ensembl.org&quot;

  r &lt;- GET(paste(server, ext, sep = &quot;&quot;), content_type(&quot;application/json&quot;))
  stop_for_status(r)

  out &lt;- as_tibble(fromJSON(toJSON(content(r)))) %&gt;% unnest()
  return(out)
}


other.snps &lt;- intersect(LD.SNP.region$variation2, dat.bmi.sel.region$SNP) ## see futher down why intersect here. 
## check this: https://www.ensembl.org/Homo_sapiens/Export/Output/Location?_format=HTML;db=core;output=ld;pop1=1000GENOMES:phase_3:PUR;r=17:50040862-50060861;v=rs199141;vdb=variation;vf=92317

LD.SNP.SNPs &lt;- purrr::map_df(other.snps, f.extract.ld, snp, &quot;EUR&quot;) %&gt;% mutate(r2 = as.numeric(r2)) %&gt;% bind_rows() %&gt;% unnest()
LD.SNP.SNPs</code></pre>
<pre><code>## # A tibble: 10 x 5
##    variation1 population_name         variation2     r2 d_prime 
##    &lt;chr&gt;      &lt;chr&gt;                   &lt;chr&gt;       &lt;dbl&gt; &lt;chr&gt;   
##  1 rs1421085  1000GENOMES:phase_3:EUR rs9939609  0.918  0.995690
##  2 rs1421085  1000GENOMES:phase_3:EUR rs9941349  0.877  0.953708
##  3 rs1421085  1000GENOMES:phase_3:EUR rs17817449 0.921  0.995704
##  4 rs6499640  1000GENOMES:phase_3:EUR rs1421085  0.0765 0.393204
##  5 rs1421085  1000GENOMES:phase_3:EUR rs11075987 0.575  0.900608
##  6 rs1421085  1000GENOMES:phase_3:EUR rs3751813  0.602  0.994207
##  7 rs1421085  1000GENOMES:phase_3:EUR rs8050136  0.918  0.995690
##  8 rs1421085  1000GENOMES:phase_3:EUR rs1558902  1      1.000000
##  9 rs1421085  1000GENOMES:phase_3:EUR rs9930506  0.819  0.916287
## 10 rs1421085  1000GENOMES:phase_3:EUR rs1121980  0.957  1.000000</code></pre>
<p>To calculate the LD matrix or the LD between SNP pairs takes a substantial amount of time.</p>
<p>BTW, ensemble has many more APIs to access, check the list <a href="https://rest.ensembl.org/documentation/">here</a>.</p>
</div>
<div id="plot-all-this" class="section level3">
<h3>Plot all this</h3>
<!---------- ------------------------------->
<p>For the locuszoomplot we need only the correlation between the top SNP and all other SNPs. So we join the object <code>LD.SNP.region</code> to <code>dat.bmi.sel.region</code>.</p>
<pre class="r"><code>dat.bmi.plot &lt;- dat.bmi.sel.region %&gt;% full_join(LD.SNP.region, by = c(&quot;SNP&quot; = &quot;variation2&quot;))</code></pre>
<pre class="r"><code>ggplot(data = dat.bmi.plot) + 
  geom_point(aes(POS, -log10(P), color = r2), shape = 16) + 
  labs(title = &quot;Locuszoomplot for BMI GWAS&quot;, subtitle = paste(&quot;Summary statistics for chromosome&quot;, sel.chr, &quot;from&quot;, format((sel.pos - range), big.mark = &quot;&#39;&quot;), &quot;to&quot;, format((sel.pos + range), big.mark = &quot;&#39;&quot;), &quot;bp&quot;), caption = paste(&quot;Data source:&quot;, url)) +
  geom_point(data = dat.bmi.plot %&gt;% filter(SNP == &quot;rs1421085&quot;), aes(POS, -log10(P)), color = &quot;black&quot;, shape = 16) + scale_color_distiller(type = &quot;div&quot;, palette = &quot;Spectral&quot;, limits = c(0,1))</code></pre>
<pre><code>## Warning: Removed 205 rows containing missing values (geom_point).</code></pre>
<p><img src="/post/2018-08-09-put-the-color-into-the-points_files/figure-html/plot-summarystats-ld-1.png" width="672" /></p>
<pre class="r"><code>#+ gghighlight(SNP == &quot;rs1421085&quot;)
# + scale_colour_gradient(limits = c(0,1))</code></pre>
<p>But wait, why are there so few SNPs in LD with thtis SNP?</p>
<!---------- ------------------------------->
</div>
</div>
<div id="things-that-do-not-work" class="section level2">
<h2>Things that do not work</h2>
<!---------- ------------------------------->
<div id="rsnps" class="section level3">
<h3><code>rsnps</code></h3>
<!------------------------------>
<p>A perfect solution would be to use the function <code>ld_search</code> from R package <a href="https://cran.r-project.org/web/packages/rsnps/vignettes/rsnps_vignette.html"><code>rsnps</code></a>. It has arguments to choose the reference panel, the population, the distance from the focal SNP… The problem is, that it only uses old reference panels (HapMap and 1KG-phase1). That means, some newer reference panel populations are left out. Besides, for some reason <code>ld_search</code> returns me an error.</p>
<pre class="r"><code># devtools::install_github(&quot;ropensci/rsnps&quot;)
# library(rsnps)
# ld_search(&quot;rs12286929&quot;) ## not working, and if, only working for phase1</code></pre>
<p><a href="https://github.com/ropensci/rsnps/issues/60" class="uri">https://github.com/ropensci/rsnps/issues/60</a></p>
</div>
<div id="api-provided-by-sph.umich" class="section level3">
<h3>API provided by sph.umich</h3>
<!------------------------------>
<p><a href="http://portaldev.sph.umich.edu/docs/api/v1/?shell#single-variant-statistics">API</a> Accessing precomputed things.</p>
<p><a href="https://github.com/statgen/locuszoom-api/issues/21#issuecomment-414434704" class="uri">https://github.com/statgen/locuszoom-api/issues/21#issuecomment-414434704</a></p>
<!------------------------------>
</div>
</div>
<div id="add-ld-to-locuszoom-plot" class="section level2">
<h2>Add LD to locuszoom plot</h2>
<!------------------------------>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 3.5.1 (2018-07-02)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS High Sierra 10.13.6
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] bindrcpp_0.2.2   xml2_1.2.0       jsonlite_1.5     httr_1.3.1      
##  [5] rsnps_0.2.6.9414 forcats_0.3.0    stringr_1.3.1    dplyr_0.7.6     
##  [9] purrr_0.2.5      readr_1.1.1      tidyr_0.8.1      tibble_1.4.2    
## [13] ggplot2_3.0.0    tidyverse_1.2.1 
## 
## loaded via a namespace (and not attached):
##  [1] tidyselect_0.2.4   xfun_0.3           haven_1.1.2       
##  [4] lattice_0.20-35    colorspace_1.3-2   htmltools_0.3.6   
##  [7] yaml_2.2.0         XML_3.98-1.13      rlang_0.2.1       
## [10] pillar_1.3.0       httpcode_0.2.0     glue_1.3.0        
## [13] withr_2.1.2        RColorBrewer_1.1-2 modelr_0.1.2      
## [16] readxl_1.1.0       bindr_0.1.1        plyr_1.8.4        
## [19] munsell_0.5.0      blogdown_0.8       gtable_0.2.0      
## [22] cellranger_1.1.0   rvest_0.3.2        evaluate_0.11     
## [25] labeling_0.3       knitr_1.20         curl_3.2          
## [28] triebeard_0.3.0    urltools_1.7.1     broom_0.5.0       
## [31] Rcpp_0.12.18       scales_0.5.0       backports_1.1.2   
## [34] hms_0.4.2          digest_0.6.15      stringi_1.2.4     
## [37] bookdown_0.7       grid_3.5.1         rprojroot_1.3-2   
## [40] cli_1.0.0          tools_3.5.1        magrittr_1.5      
## [43] lazyeval_0.2.1     crul_0.6.0         crayon_1.3.4      
## [46] pkgconfig_2.0.1    data.table_1.11.4  lubridate_1.7.4   
## [49] assertthat_0.2.0   rmarkdown_1.10     rstudioapi_0.7    
## [52] R6_2.2.2           icon_0.1.0         nlme_3.1-137      
## [55] compiler_3.5.1</code></pre>
</div>

</div>

        <footer class="post-footer clearfix">
    
        <p class="post-tags">
            <span>Tagged:</span>
            
            
                <a href="/tags/statgen/">statgen</a>, 
            
                <a href="/tags/r/">R</a>, 
            
                <a href="/tags/data-visualisation/">data visualisation</a>, 
            
                <a href="/tags/r/">R</a>, 
            
                <a href="/tags/ggplot2/">ggplot2</a>
            
        </p>
    

    <div class="share">
        

        

        
        
    </div>
</footer>

        
    <div class="comments">
        <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "sinarueeger-github-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
    </div>

    </article>
</div>

    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
      tex2jax: {
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
      }
    });
    </script>
    <script type="text/javascript"
      src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
    </script>

    <footer class="footer">
      <div class="container">
        <p class="footer-copyright">
                    <span>&copy; 2018. All rights reserved.</span>
        </p>
      </div>
    </footer>
    
  </body>
</html>

