<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <title>Locuszoom plot of GWAS summary statistics &middot; Sina Rüeger</title>
        <meta name="description" content="using the `biomaRt` R-package">
        <meta name="HandheldFriendly" content="True">
        <meta name="MobileOptimized" content="320">
        <meta name="generator" content="Hugo 0.38.2" />
        <meta name="robots" content="index,follow">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        <link rel="stylesheet" href="/dist/styles.css">
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Open+Sans:300italic,400italic,600italic,700italic,400,600,700,300&subset=latin,cyrillic-ext,latin-ext,cyrillic">
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous">
        
        
    </head>
    <body>
        

        <div id="wrapper">
            <header class="site-header">
                <div class="container">
                    <div class="site-title-wrapper">
                        
                            <h1 class="site-title">
                                <a title="Analysing genetic data" href="/">Analysing genetic data</a>
                            </h1>
                        
                        <a class="button-square" href="/index.xml"><i class="fa fa-rss"></i></a>
                        
                            <a class="button-square button-social hint--top" data-hint="Twitter" title="Twitter" href="https://twitter.com/sinarueeger">
                                <i class="fa fa-twitter"></i>
                            </a>
                        
                        
                        
                            <a class="button-square button-social hint--top" data-hint="Github" title="Github" href="https://github.com/sinarueeger">
                                <i class="fa fa-github-alt"></i>
                            </a>
                        
                        
                        
                            <a class="button-square button-social hint--top" data-hint="LinkedIn" title="LinkedIn" href="https://www.linkedin.com/in/sinarueeger/">
                                <i class="fa fa-linkedin"></i>
                            </a>
                        
                        
                            <a class="button-square button-social hint--top" data-hint="Google+" title="Google+" href="https://scholar.google.com/citations?view_op=list_works&amp;hl=de&amp;authuser=1&amp;user=OD9gCh4AAAAJ">
                                <i class="fa fa-google-plus"></i>
                            </a>
                        
                        
                            <a class="button-square button-social hint--top" data-hint="Email" title="Email" href="mailto:sina.rueeger@gmail.com">
                                <i class="fa fa-envelope"></i>
                            </a>
                        
                    </div>

                    <ul class="site-nav">
                        
    <li class="site-nav-item">
        <a title="Blog" href="/">Blog</a>
    </li>

    <li class="site-nav-item">
        <a title="Resources" href="https://github.com/sinarueeger/statistical-genetics-resources/blob/master/statgen-data.md">Resources</a>
    </li>

    <li class="site-nav-item">
        <a title="About" href="/page/about/">About</a>
    </li>

    <li class="site-nav-item">
        <a title="Contact" href="/page/contact/">Contact</a>
    </li>

                    </ul>
                </div>
            </header>

            <div id="container">


<div class="container">
    <article class="post-container" itemscope="" itemtype="http://schema.org/BlogPosting">
        <header class="post-header">
    <h1 class="post-title" itemprop="name headline">Locuszoom plot of GWAS summary statistics</h1>
    
        <p class="post-description" itemprop="description">using the `biomaRt` R-package</p>
    
    <p class="post-date">
        <span>Published <time datetime="2018-07-23" itemprop="datePublished">Mon, Jul 23, 2018</time></span>
        <span>by</span>
        <span itemscope="" itemprop="author" itemtype="https://schema.org/Person">
            <span itemprop="name">
                <a href="https://twitter.com/sinarueeger" itemprop="url" rel="author">Sina Rüeger</a>
            </span>
        </span>
    </p>
</header>

        <div class="post-content clearfix" itemprop="articleBody">
    

    <link href="/rmarkdown-libs/font-awesome/css/fontawesome-all.min.css" rel="stylesheet" />


<!-- what do we want -->
<p>In the world of <a href="https://www.ebi.ac.uk/training/online/course/gwas-catalog-exploring-snp-trait-associations/why-do-we-need-gwas-catalog/what-are-genome">genome-wide association studies</a> (GWAS) we often get a list of genetic markers (<a href="https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">SNPs</a>) that seem for some reason relevant for a particular outcome. At the same time, we have little knowledge about these genetic variants that come in cryptic combinations of characters and numbers.</p>
<p><em>For example:</em></p>
<ul>
<li><em>We might get asked how frequent the SNP <code>rs1421085</code> is in a certain population. </em></li>
<li><em>We need to extract all known SNPs that are within 1 Mb of <code>rs1421085</code>. </em></li>
<li><em>A genomic region turns out to be highly relevant for some disease and we want to know all genes contained in that genomic region.</em></li>
</ul>
<p>The list of SNPs come typically from summarised data. Minimally, these datasets contain the SNP identifier (e.g. <code>rs1421085</code>), the effect size (<code>b</code>) and the standard error (<code>se</code>). Sometimes the position (<code>POS</code>) and chromosome (<code>CHR</code>) is provided instead of the SNP identifier, and sometimes both is available. Then there is usually other information coming from the study data, for example the allele frequency in the study. However, hardly ever do the datasets contain annotation, such as the gene where the SNP resides or the allele frequency in a certain population.</p>
<p>This makes sense. Genetic data from cohorts will be used for years and won’t change, but the annotation will change over the years. Take the the chromosomal position of SNPs, that changes every couple of years.</p>
<p>To get our hands on annotation, we need to consult external databases that are - lucky us! - public. But first, let us define what we want to see at the end of the blogpost.</p>
<div id="goal" class="section level2">
<h2>Goal</h2>
<p>For this blogpost, we keep it simple and only focus on the <strong>annotation of a genomic region</strong>, but this can be easily extended to any other annotation.</p>
<!-- ### Annotation of each SNP

We might want to know the corresponding gene of a SNP.

<figure>
<img src="/post/annotation/table-drawing.jpg" alt="Drawing of a table." align="middle" style="width: 400px;"/>
<figcaption><small>Here, we know the SNP identifier, the chromosome and the position form the GWAS results, but not the corresponding gene.</small></figcaption>
</figure> -->
<p>In the illustration below, we want to know the gene starting and end positions around the FTO gene region to create an informative locuszoom plot.</p>
<figure>
<img src="/post/annotation/locuszoom-drawing.jpg" alt="Drawing of a table." align="middle" style="width: 400px;"/>
<figcaption>
<small>In this case, we want to visualise the GWAS summary statistics of a genomic region (with each point representing a SNP), with the corresponding genes at the bottom. This is similar to a plot done with <a href="http://locuszoom.org/">LocusZoom</a>, a tool that takes summary statistics as input and outputs a pretty graph for a desired genomic region, including gene annotation, LD information and more. </small>
</figcaption>
</figure>
</div>
<div id="getting-it-done" class="section level2">
<h2>Getting it done</h2>
<!-- how can we achieve this? -->
<p>There are currently more than 10 Mio SNPs known, and knowing their functions and genes by heart would equal to some super power. Which is why we <i class="fas  fa-heart " style="color:#a6cee3;"></i><a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a> public databases, such as <a href="https://www.ncbi.nlm.nih.gov/projects/SNP/">dbSNP</a> and <a href="https://www.ensembl.org/index.html">ensembl</a><a href="#fn2" class="footnoteRef" id="fnref2"><sup>2</sup></a>.</p>
<p>What we <i class="fas  fa-heart " style="color:#377eb8;"></i> even more, is, to make our analyses reproducible and automate annotation and lookups.</p>
<p><i class="fas  fa-arrow-right " style="color:black;"></i> In this blogpost, I will show how to zoom into GWAS results and annotate the plot based on the information about that genomic region using <i class="fab  fa-r-project "></i>.</p>
<!-- why do we need a locus zoom plot anyway ?-->
<!-- data -->
</div>
<div id="get-summary-statistics" class="section level2">
<h2>1. Get summary statistics</h2>
<p>First, we need some GWAS summary statistics.</p>
<p>There are lots of <a href="https://github.com/sinarueeger/statistical-genetics-resources/blob/master/statgen-data.md#gwas-summary-statistic-results"><strong>resources</strong></a> for publicly available GWAS summary statistics.</p>
<p>We will look at BMI, because it can be accessed easily<a href="#fn3" class="footnoteRef" id="fnref3"><sup>3</sup></a>. The data is from the <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files">Genetic Investigation of ANthropometric Traits</a> (GIANT) consortium. You can <strong>download</strong> the dataset<a href="#fn4" class="footnoteRef" id="fnref4"><sup>4</sup></a> <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/GIANT_consortium_data_files#GWAMA_Age-.2FSex-Stratified_2015_BMI_and_WHR">here</a> or - even better - load it <strong>directly into R</strong>.</p>
<pre class="r"><code>## Data Source URL
url &lt;- &quot;https://portals.broadinstitute.org/collaboration/giant/images/2/21/BMI_All_ancestry.fmt.gzip&quot;

## Import BMI summary statistics
## dat.bmi &lt;- read_tsv(file = url) ## taking too long, let&#39;s use fread instead.

dat.bmi &lt;- data.table::fread(url, verbose = FALSE)</code></pre>
<p>I added <code>verbose = FALSE</code> because it will complain that there is an unexpected character in column 1, which appears to be numerical. This is because chromosome X will only appear towards the end of the dataset.</p>
<p>Next, we rename some columns to something more conventional.</p>
<pre class="r"><code>## Rename some columns
dat.bmi &lt;- dat.bmi %&gt;% rename(SNP = SNPNAME, P = Pvalue)</code></pre>
<p>Now, let’s look at the data.</p>
<pre class="r"><code>skimr::skim(dat.bmi)</code></pre>
<pre><code>## Skim summary statistics
##  n obs: 246328 
##  n variables: 10 
## 
## Variable type: character 
##  variable missing complete      n min max empty n_unique
##       ALT       0   246328 246328   1   1     0        4
##  ExAC_MAF       0   246328 246328   1 325     0    51423
##      GMAF       0   246328 246328   1  17     0     9964
##       REF       0   246328 246328   1   1     0        4
##       SNP       0   246328 246328   1  11     0   243206
## 
## Variable type: integer 
##  variable missing complete      n    mean      sd    p0     p25     p50
##       CHR    5204   241124 246328 9.57    6.3         1 4       9      
##       POS       0   246328 246328 7.6e+07 5.7e+07 11885 3.2e+07 6.1e+07
##       p75     p100     hist
##  15       22       ▇▅▃▅▃▂▅▂
##   1.1e+08  2.5e+08 ▇▇▅▅▃▂▁▁
## 
## Variable type: numeric 
##  variable missing complete      n      mean     sd        p0    p25
##      beta     977   245351 246328   0.00051   0.13 -3.2      -0.032
##         P     977   245351 246328   0.48      0.3   8.6e-269  0.21 
##        se       0   246328 246328 Inf       NaN     0.002     0.023
##       p50   p75  p100     hist
##  -0.00017 0.031   3.2 ▁▁▁▇▇▁▁▁
##   0.47    0.73    1   ▇▇▆▆▆▆▆▆
##   0.059   0.11  Inf   ▇▁▁▁▁▁▁▁</code></pre>
<p>The reference allele (<code>REF</code>), alternative allele (<code>ALT</code>), SNP identifier (<code>SNP</code>) and chromosome (<code>CHR</code>) are characters. There are four unique values for <code>ALT</code> and <code>REF</code>: A, C, G, T, and 23 unique values for <code>CHR</code>. The two columns with the minor allele frequencies measured in <a href="https://portals.broadinstitute.org/collaboration/giant/index.php/Main_Page">GIANT</a> and <a href="http://exac.broadinstitute.org/">ExAC</a> datasets (<code>GMAF</code>, <code>ExAC_MAF</code>) are characters too, because the allele is concatenated. The chromosomal position (<code>POS</code>) is an integer. Then there is the actual association of each SNP with BMI (<code>beta</code>, <code>se</code>, <code>P</code>).</p>
<!-- The first column indicates the chromosome (`CHR`), the second the chromosomal position (`POS`), then the reference allele (`REF`), the alternative allele (`ALT`), the SNP identifier (`SNP`), minor allele frequencies measured in [GIANT](https://portals.broadinstitute.org/collaboration/giant/index.php/Main_Page) and [ExAC](http://exac.broadinstitute.org/) datasets (`GMAF`, `ExAC_MAF`), followed by the association of that SNP with BMI (`beta`, `se`, `P`). -->
<p>This dataset has 246328 rows and 10 columns. What if we want to visualise this all at once? In particularly, we are interested if there are ANY associations between the genetic markers and BMI. The mini <code>P</code> distribution in the skimr output does not reveal much.</p>
<div id="visualising-associations" class="section level3">
<h3>Visualising associations</h3>
<p>Visualising all associations at once can be done with a Manhatten plot (creating an immediate <i class="fas  fa-smile " style="color:gray;"></i> or <i class="fas  fa-frown " style="color:gray;"></i> effect), where the x-axis represents chromosome<code>CHR</code> and the chromosomal position <code>POS</code>, and the y-axis the <code>-log10(P)</code>-value. Let’s use the R-package <a href="https://github.com/drveera/ggman"><code>ggman</code></a> by Veera Manikandan for that.</p>
<pre class="r"><code>## ggman installed via gh: https://github.com/drveera/ggman
ggman::ggman(dat.bmi, snp = &quot;SNP&quot;, bp = &quot;POS&quot;, chrom = &quot;CHR&quot;, pvalue = &quot;P&quot;)</code></pre>
<p><img src="/post/2018-07-04-plotting-local-genomic-association-results-in-r_files/figure-html/mhtplot-1.png" width="672" /></p>
<p>We can spot immediately, that there are loads of SNPs with P-values smaller than <span class="math display">\[10^{-100}\]</span> (sample size was around <a href="https://www.biorxiv.org/content/early/2018/03/22/274654">700K</a>).</p>
<!-- ```{r, qqplot}
library(cookbookRsina)
QQplot(dat.bmi$P)

``` --->
<p>Of course, there are many other (and probably better) solutions to spot real associations, but this is not the point of this blogpost ;-).</p>
</div>
<div id="identify-genomic-region-with-lowest-p-value" class="section level3">
<h3>Identify genomic region with lowest P-value</h3>
<p>Now that we know that there are lots of genetic markers associated with BMI, we want to look at a specific genomic region and figure out what genes it contains. So let’s pick the genomic region with the lowest P-value.</p>
<pre class="r"><code>dat.bmi.sel &lt;- dat.bmi %&gt;% slice(which.min(P))
dat.bmi.sel</code></pre>
<pre><code>## # A tibble: 1 x 10
##     CHR      POS REF   ALT   SNP     GMAF  ExAC_MAF  beta     se         P
##   &lt;int&gt;    &lt;int&gt; &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt;     &lt;dbl&gt;
## 1    16 53800954 T     C     rs1421… C:0.… -        0.078 0.0022 8.60e-269</code></pre>
<p>SNP identifier rs1421085 that has the lowest P-value (P = 8.610^{-269}).</p>
<p>Now we can visualise the summary statistics of that genomic region (<span class="math inline">\(\pm\)</span> 500 x 10^{3}).</p>
<pre class="r"><code>range &lt;- 500e3
sel.chr &lt;- dat.bmi.sel$CHR
sel.pos &lt;- dat.bmi.sel$POS

dat.bmi.sel.region &lt;- dat.bmi %&gt;% filter(CHR == sel.chr, between(POS, sel.pos - range, sel.pos + range))</code></pre>
<pre class="r"><code>p1 &lt;- ggplot(data = dat.bmi.sel.region) + geom_point(aes(POS, -log10(P))) + labs(title = &quot;Locuszoomplot for BMI GWAS&quot;,
       subtitle = paste(&quot;Chromosome&quot;, sel.chr, &quot;from&quot;, format((sel.pos - range), big.mark = &quot;&#39;&quot;) , &quot;to&quot;, format((sel.pos + range), big.mark = &quot;&#39;&quot;), &quot;bp&quot; ), caption = paste(&quot;Data source:&quot;, url))
print(p1)</code></pre>
<p><img src="/post/2018-07-04-plotting-local-genomic-association-results-in-r_files/figure-html/plot-summarystats-1.png" width="672" /></p>
<p>Next, we want to know if rs1421085 is part of a gene, and if yes, which one.</p>
</div>
</div>
<div id="extracting-annotation" class="section level2">
<h2>2. Extracting annotation</h2>
<div id="biomart" class="section level3">
<h3>biomaRt</h3>
<p>Thankfully, there is an R-package called <a href="https://github.com/ropensci/biomartr"><code>biomaRt</code></a> that can do this for us.</p>
<p>I have only come across this package a few weeks ago, so I apologise in advance that I won’t make full use of all the features the package offers.</p>
<p><i class="fas  fa-exclamation "></i> <code>biomaRt</code> should not to be confused with <code>biomartr</code>. <code>biomartr</code> is is an rOpenSci package by <a href="https://twitter.com/HajkDrost">Hajk-Georg Drost</a>. I needed this tweet <i class="fas  fa-arrow-down " style="color:black;"></i> to realise that these were two separate, yet related R-packages.</p>
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr">
Did you ever want to reproducibly retrieve thousands of genomes across the tree of life using only one R command? Then have a look at the new version of biomartr which is on its way to CRAN! <a href="https://t.co/kWF5XCoGhj">https://t.co/kWF5XCoGhj</a> <a href="https://twitter.com/hashtag/bioinformatics?src=hash&amp;ref_src=twsrc%5Etfw">#bioinformatics</a> <a href="https://twitter.com/hashtag/rstats?src=hash&amp;ref_src=twsrc%5Etfw">#rstats</a> <a href="https://twitter.com/hashtag/Genomics?src=hash&amp;ref_src=twsrc%5Etfw">#Genomics</a> <a href="https://t.co/EZHJP0n1f9">pic.twitter.com/EZHJP0n1f9</a>
</p>
— Hajk-Georg Drost (<span class="citation">@HajkDrost</span>) <a href="https://twitter.com/HajkDrost/status/1012280903488466945?ref_src=twsrc%5Etfw">June 28, 2018</a>
</blockquote>
<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
<p>So far, I did not fully grasp the benefits of <code>biomartr</code> compared to <code>biomaRt</code> for annotation of human data. But I will definitely <a href="https://ropensci.github.io/biomartr/articles/Functional_Annotation.html">look more into it</a>.</p>
</div>
<div id="learning-resources" class="section level3">
<h3>Learning resources</h3>
<p>I used mainly two sources to learn what I wanted to do:</p>
<ul>
<li><a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/biomaRt/inst/doc/biomaRt.html">Vignette</a></li>
<li><a href="https://stackoverflow.com/questions/41490657/best-way-to-get-list-of-snps-by-gene-id">A question on StackOverflow</a></li>
</ul>
</div>
<div id="using-biomart" class="section level3">
<h3>Using biomaRt</h3>
<p>First we need to load the <code>biomaRt</code> <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html">bioconductor package</a>.</p>
<pre class="r"><code>library(biomaRt) ## </code></pre>
<p>Next we specify which database to use. You can use the functions <code>listMart()</code>, <code>listEnsembl</code> and <code>listDatasets()</code> to select the right biomart and dataset. We will need to extract SNPs and genes.</p>
<pre class="r"><code>snp.ensembl &lt;- useEnsembl(biomart=&quot;snp&quot;, dataset=&quot;hsapiens_snp&quot;)
gene.ensembl &lt;- useEnsembl(biomart=&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)

# other ways of selecting the mart
# snp.mart &lt;- useMart(biomart = &quot;ENSEMBL_MART_SNP&quot;, dataset=&quot;hsapiens_snp&quot;)
# gene.mart &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)</code></pre>
<p>Last, we . To check which attributes and filters are available, check functions <code>listAttributes(snp.ensembl)</code> and <code>listFilters(snp.ensembl)</code>.</p>
<pre class="r"><code>out.bm &lt;- getBM(attributes = c(
  &quot;ensembl_gene_stable_id&quot;,
  &#39;refsnp_id&#39;,
  &#39;chr_name&#39;,
  &#39;chrom_start&#39;,
  &#39;chrom_end&#39;,
  &#39;minor_allele&#39;,
  &#39;minor_allele_freq&#39;),
  filters = &#39;snp_filter&#39;,
  values = dat.bmi.sel$SNP,
  mart = snp.ensembl
)
out.bm</code></pre>
<pre><code>##   ensembl_gene_stable_id refsnp_id chr_name chrom_start chrom_end
## 1        ENSG00000140718 rs1421085       16    53767042  53767042
##   minor_allele minor_allele_freq
## 1            C          0.228634</code></pre>
<p>Before getting the gene names, I want to check if my chromosome and position in the dataset are actually matching with the info in the database. This is a handy thing, because oftentimes, only SNPs are reported. Or SNPs are reported, but with positions on a different build.</p>
<p>This gives us - as choosen in <code>attributes</code> - the gene identifier, the SNP identifier, the chromosome name, chromsomal position (start + end), minor allele and minor allele frequency. This corresponds to this webpage entry on the <a href="https://www.ensembl.org/Homo_sapiens/Variation/Explore?r=16:53766542-53767542;v=rs1421085;vdb=variation;vf=984887">ensembl webpage</a>.</p>
<pre class="r"><code>ifelse(dat.bmi.sel$CHR == out.bm$chr_name, &quot;chromosome matching&quot;, &quot;chromosome is not matching&quot;)</code></pre>
<pre><code>## [1] &quot;chromosome matching&quot;</code></pre>
<pre class="r"><code>ifelse(dat.bmi.sel$POS == out.bm$chrom_start, &quot;position matching&quot;, &quot;position is not matching&quot;)</code></pre>
<pre><code>## [1] &quot;position is not matching&quot;</code></pre>
<p>Next, we want to get the gene name.</p>
<p>Let’s check which <code>attributes</code> contain <code>gene</code>.</p>
<pre class="r"><code>listAttributes(snp.ensembl) %&gt;% slice(str_which(description, &quot;Gene&quot;)) </code></pre>
<pre><code>## # A tibble: 1 x 3
##   name                   description    page 
##   &lt;chr&gt;                  &lt;chr&gt;          &lt;chr&gt;
## 1 ensembl_gene_stable_id Gene stable ID snp</code></pre>
<pre class="r"><code>## extract gene-d
## ----------
out.bm.snpmart &lt;- getBM(attributes = c(&#39;refsnp_id&#39;,&#39;allele&#39;,&#39;chrom_start&#39;,&#39;chrom_strand&#39;, &#39;chr_name&#39;, &#39;phenotype_description&#39;, &#39;ensembl_gene_stable_id&#39;,&#39;associated_gene&#39;), 
      filters = c(&#39;snp_filter&#39;), 
      values = dat.bmi.sel$SNP, 
      mart = snp.ensembl)
out.bm.snpmart</code></pre>
<pre><code>##   refsnp_id allele chrom_start chrom_strand chr_name
## 1 rs1421085    T/C    53767042            1       16
## 2 rs1421085    T/C    53767042            1       16
## 3 rs1421085    T/C    53767042            1       16
## 4 rs1421085    T/C    53767042            1       16
## 5 rs1421085    T/C    53767042            1       16
## 6 rs1421085    T/C    53767042            1       16
## 7 rs1421085    T/C    53767042            1       16
##                phenotype_description ensembl_gene_stable_id
## 1                    Type 2 diabetes        ENSG00000140718
## 2 OBESITY (BMIQ14) SUSCEPTIBILITY TO        ENSG00000140718
## 3                    Body Mass Index        ENSG00000140718
## 4      Obesity (early onset extreme)        ENSG00000140718
## 5                            Obesity        ENSG00000140718
## 6       Dietary macronutrient intake        ENSG00000140718
## 7          Childhood body mass index        ENSG00000140718
##   associated_gene
## 1             FTO
## 2             FTO
## 3                
## 4             FTO
## 5             FTO
## 6             FTO
## 7             FTO</code></pre>
<pre class="r"><code>## extract name
## ----------
ensembl &lt;- useMart(&quot;ensembl&quot;,dataset=&quot;hsapiens_gene_ensembl&quot;)
out.bm.ensembl &lt;- getBM(attributes = c(&#39;external_gene_name&#39;), 
      filters = c(&#39;ensembl_gene_id&#39;),
      values = unique(out.bm.snpmart$ensembl_gene_stable_id), 
      mart = ensembl)
out.bm.ensembl</code></pre>
<pre><code>##   external_gene_name
## 1                FTO</code></pre>
<p>The gene is called FTO.</p>
</div>
<div id="extracting-gene-names-for-genomic-region" class="section level3">
<h3>Extracting gene names for genomic region</h3>
<p>So now we know that rs1421085 is part of FTO. But where does FTO start and end? And what are the genes nearby? For this purpose, we want to visualise the summary statistics of the full genomic region (<span class="math inline">\(\pm 250e3\)</span>).</p>
<pre class="r"><code>listAttributes(ensembl) %&gt;% slice(str_which(description, &quot;Gene type&quot;)) </code></pre>
<pre><code>## # A tibble: 3 x 3
##   name         description page        
##   &lt;chr&gt;        &lt;chr&gt;       &lt;chr&gt;       
## 1 gene_biotype Gene type   feature_page
## 2 gene_biotype Gene type   structure   
## 3 gene_biotype Gene type   sequences</code></pre>
<pre class="r"><code>listFilters(ensembl)%&gt;% slice(str_which(description, &quot;Gene&quot;)) </code></pre>
<pre><code>## # A tibble: 121 x 2
##    name                       description                                 
##    &lt;chr&gt;                      &lt;chr&gt;                                       
##  1 with_entrezgene_trans_name With EntrezGene transcript name ID(s)       
##  2 with_genedb                With GeneDB ID(s)                           
##  3 with_uniprot_gn            With UniProtKB Gene Name ID(s)              
##  4 with_wikigene              With WikiGene ID(s)                         
##  5 ensembl_gene_id            Gene stable ID(s) [e.g. ENSG00000000003]    
##  6 ensembl_gene_id_version    Gene stable ID(s) with version [e.g. ENSG00…
##  7 external_gene_name         Gene Name(s) [e.g. RF00100]                 
##  8 entrezgene_trans_name      EntrezGene transcript name ID(s) [e.g. C2or…
##  9 genedb                     GeneDB ID(s) [e.g. Smp_009580.1:pep]        
## 10 uniprot_gn                 UniProtKB Gene Name ID(s) [e.g. A0A023T6R1] 
## # ... with 111 more rows</code></pre>
<pre class="r"><code>out.bm.genes.region &lt;- getBM(attributes = c(&#39;start_position&#39;,&#39;end_position&#39;,&#39;ensembl_gene_id&#39;,&#39;external_gene_name&#39;, &#39;gene_biotype&#39;), 
      filters = c(&#39;chromosome_name&#39;,&#39;start&#39;,&#39;end&#39;), 
      values = list(sel.chr, sel.pos - range, sel.pos + range), 
      mart = ensembl)
out.bm.genes.region</code></pre>
<pre><code>##    start_position end_position ensembl_gene_id external_gene_name
## 1        53055033     53329150 ENSG00000177200               CHD9
## 2        53310797     53311989 ENSG00000279589         AC079416.2
## 3        53334562     53334664 ENSG00000238645            RF01210
## 4        53337453     53337571 ENSG00000202193          RNA5SP427
## 5        53362019     53363678 ENSG00000259962         AC007342.2
## 6        53364982     53373083 ENSG00000260078         AC007342.3
## 7        53373493     53384745 ENSG00000261804         AC007342.4
## 8        53386944     53389085 ENSG00000262714         AC007342.5
## 9        53389886     53390563 ENSG00000279741         AC007342.8
## 10       53391089     53392473 ENSG00000280027         AC007342.9
## 11       53433977     53491649 ENSG00000103479               RBL2
## 12       53443228     53443332 ENSG00000252569         RNU6-1153P
## 13       53448748     53449590 ENSG00000259926         AC007342.1
## 14       53478957     53481550 ENSG00000279344         AC007342.6
## 15       53487607     53489943 ENSG00000279722         AC007342.7
## 16       53491040     53504411 ENSG00000166971              AKTIP
## 17       53598153     53703938 ENSG00000103494           RPGRIP1L
## 18       53628256     53628816 ENSG00000275191         AC007497.1
## 19       53701692     54158512 ENSG00000140718                FTO
## 20       53981229     53981912 ENSG00000261630         AC007496.2
## 21       53991012     53994454 ENSG00000280392         AC007496.3
## 22       53998313     53999969 ENSG00000260194         AC007496.1
## 23       54033997     54054583 ENSG00000261049         AC007347.1
## 24       54114697     54114878 ENSG00000280454         AC007347.2
## 25       54239693     54240654 ENSG00000259759         AC007907.1
## 26       54245618     54270746 ENSG00000259517          LINC02169
## 27       54283304     54286763 ENSG00000177508               IRX3
## 28       54285604     54289568 ENSG00000283689         AC018553.2
## 29       54290965     54292422 ENSG00000277559         AC018553.1
##                          gene_biotype
## 1                      protein_coding
## 2                                 TEC
## 3                              snoRNA
## 4                                rRNA
## 5              unprocessed_pseudogene
## 6  transcribed_unprocessed_pseudogene
## 7                             lincRNA
## 8                             lincRNA
## 9                                 TEC
## 10                                TEC
## 11                     protein_coding
## 12                              snRNA
## 13                     sense_intronic
## 14                                TEC
## 15                                TEC
## 16                     protein_coding
## 17                     protein_coding
## 18                     sense_intronic
## 19                     protein_coding
## 20                          antisense
## 21                                TEC
## 22                          antisense
## 23                          antisense
## 24                         pseudogene
## 25                            lincRNA
## 26                            lincRNA
## 27                     protein_coding
## 28                          antisense
## 29                            lincRNA</code></pre>
<pre class="r"><code>## rank gene names according to start position

out.bm.genes.region &lt;- out.bm.genes.region %&gt;% mutate(external_gene_name = fct_reorder(external_gene_name, start_position, .desc = TRUE))

## plot
p2 &lt;- ggplot(data = out.bm.genes.region) + geom_linerange(aes(x = external_gene_name, ymin = start_position, ymax = end_position)) + coord_flip() + ylab(&quot;&quot;) #+ xlim(fto.pos + c(-1,1) * range)
print(p2)</code></pre>
<p><img src="/post/2018-07-04-plotting-local-genomic-association-results-in-r_files/figure-html/plot-genes-1.png" width="672" /></p>
<p>That could be nicer</p>
<pre class="r"><code>## define plot range
plot.range &lt;- c(min(sel.pos - range, out.bm.genes.region$start_position), max(sel.pos + range, out.bm.genes.region$end_position))

## rank gene_biotype label
out.bm.genes.region &lt;- out.bm.genes.region %&gt;% mutate(gene_biotype_fac = as.factor(gene_biotype), external_gene_name = fct_reorder2(external_gene_name, start_position, gene_biotype_fac, .desc = TRUE))


## plot
p2 &lt;- ggplot(data = out.bm.genes.region) + geom_linerange(aes(x = external_gene_name, ymin = start_position, ymax = end_position, color = gene_biotype, group = gene_biotype)) + coord_flip() + ylab(&quot;&quot;) + ylim(plot.range) + geom_text(aes(x = external_gene_name, y = start_position, label = external_gene_name, color = gene_biotype), alpha = I(0.5), hjust = &quot;right&quot;, size= 2) + labs(title = &quot;&quot;,
       subtitle = paste(&quot;Genes&quot;), caption = paste(&quot;Data source:&quot;, ensembl@host, &quot;/&quot;, ensembl@dataset))+ theme(axis.title.y=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(), ## remove ticks axis
        strip.text.y = element_text(angle = 0),
        legend.position=&quot;bottom&quot;, 
        panel.grid.major.y = element_blank()) ## remove horizontal lines


print(p2)</code></pre>
<p><img src="/post/2018-07-04-plotting-local-genomic-association-results-in-r_files/figure-html/plot-genes-pretty-1.png" width="672" /></p>
<p>There are some genes starting with <em>AC-</em> and <em>LINCO</em> and <em>RINU</em>. It is tempting to remove them, since they are so small. So let’s <a href="https://www.genenames.org/about/guidelines">check</a> what they are.</p>
</div>
<div id="nice-to-have" class="section level3">
<h3>Nice to have</h3>
<p>What would be great, is to add information about the correlation between SNPs, like in <a href="http://locuszoom.org/">LocusZoom plots</a>.</p>
<p>Or even better, an interactive plot. For example, hovering over the SNP points would light up the corresponding genes or give some other information, like the allele frequency.</p>
<p>That seems like some nice future blogposts :-).</p>
<!-- GWAS + biomart info -->
</div>
</div>
<div id="combining-summary-statistics-and-annotation" class="section level2">
<h2>3. Combining summary statistics and annotation</h2>
<pre class="r"><code>library(patchwork)
p1b &lt;- p1 + xlab(&quot;&quot;) + theme(axis.title.x=element_blank(), axis.text.x=element_blank(), axis.ticks.x=element_blank()) + xlim(plot.range)

p1b + p2 + plot_layout(ncol = 1, heights = c(3, 1))</code></pre>
<p><img src="/post/2018-07-04-plotting-local-genomic-association-results-in-r_files/figure-html/summarystats-genes-1.png" width="672" /></p>
</div>
<div id="source" class="section level2">
<h2>Source</h2>
<ul>
<li>The RMarkdowns file is <a href="https://github.com/sinarueeger/webpage/blob/master/blogdown/content/post/2018-07-04-plotting-local-genomic-association-results-in-r.Rmd">here</a>.</li>
<li>Some more <code>biomaRt</code> code snippets are in a <a href="https://gist.github.com/sinarueeger/4430492e67639fccaeacf4340f7cb7df">gist</a>.</li>
</ul>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>Awesome icons in R? <i class="fas  fa-band-aid " style="color:orange;"></i> <i class="fas  fa-chess-queen " style="color:gray;"></i> <i class="fas  fa-bicycle " style="color:green;"></i> <i class="fas  fa-birthday-cake " style="color:pink;"></i> <i class="fas  fa-arrow-right " style="color:black;"></i> checkout <a href="https://ropensci.org/technotes/2018/05/15/icon/">these instructions by Mitchell O’Hara-Wild on the rOpenSci webpage</a>.<a href="#fnref1">↩</a></p></li>
<li id="fn2"><p>Thanks to my biology colleagues for explaining me the differences between these databases and what a biomart is!<a href="#fnref2">↩</a></p></li>
<li id="fn3"><p>Recently, there have been two new studies by the <a href="https://www.med.unc.edu/pgc/">Psychiatric Genomics Consortium</a> (PGC) on <a href="https://www.sciencedirect.com/science/article/pii/S0092867418306585?via%3Dihub">schizophrenia (SCZ) and bipoloar disorder (BD)</a>, as well as <a href="https://www.nature.com/articles/s41588-018-0090-3">Major depressive disorder (MDD)</a>. For all studies of PGC, summary statistics can be downloaded <a href="https://www.med.unc.edu/pgc/results-and-downloads/downloads">here</a>. For example, if you would like to download the latest MDD results, search for <em>‘MDD2 2018’</em>, for BD and SCZ search for <em>‘BIP and SCZ results from Cell Publication, 2018’</em>. Anybody can download these results, but need to acknowledge and agree to a list of conditions - which I think is an excellent approach!<a href="#fnref3">↩</a></p></li>
<li id="fn4"><p>Data source: <a href="https://www.biorxiv.org/content/early/2018/03/22/274654">Yengo et al. (2018)</a>.<a href="#fnref4">↩</a></p></li>
</ol>
</div>

</div>

        <footer class="post-footer clearfix">
    
        <p class="post-tags">
            <span>Tagged:</span>
            
            
                <a href="/tags/statgen/">statgen</a>, 
            
                <a href="/tags/r/">R</a>, 
            
                <a href="/tags/data-visualisation/">data visualisation</a>, 
            
                <a href="/tags/r/">R</a>, 
            
                <a href="/tags/ggplot2/">ggplot2</a>
            
        </p>
    

    <div class="share">
        

        

        
        
    </div>
</footer>

        
    <div class="comments">
        <div id="disqus_thread"></div>
<script>
    var disqus_config = function () {
    
    
    
    };
    (function() {
        if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
            document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
            return;
        }
        var d = document, s = d.createElement('script'); s.async = true;
        s.src = '//' + "sinarueeger-github-io" + '.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
    </div>

    </article>
</div>

            </div>
        </div>

        <footer class="footer">
            <div class="container">
                <div class="site-title-wrapper">
                    <h1 class="site-title">
                        <a title="Analysing genetic data" href="/">Analysing genetic data</a>
                    </h1>
                    <a class="button-square button-jump-top js-jump-top" href="#">
                        <i class="fa fa-angle-up"></i>
                    </a>
                </div>

                <p class="footer-copyright">
                    <span>&copy; 2018 / Powered by <a href="https://gohugo.io/">Hugo</a></span>
                </p>
                <p class="footer-copyright">
                    <span><a href="https://github.com/roryg/ghostwriter">Ghostwriter theme</a> By <a href="http://jollygoodthemes.com">JollyGoodThemes</a></span>
                    <span>/ <a href="https://github.com/jbub/ghostwriter">Ported</a> to Hugo By <a href="https://github.com/jbub">jbub</a></span>
                </p>
            </div>
        </footer>

        <script src="/js/jquery-1.11.3.min.js"></script>
        <script src="/js/jquery.fitvids.js"></script>
        
        
            <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        
        
        <script src="/js/scripts.js"></script>
    </body>
</html>

